byline.go

```diff
@@ -76,6 +79,9 @@ func (lr *Reader) scanLinesBySep(data []byte, atEOF bool) (advance int, token []
 
 // Read - implement io.Reader interface
 func (lr *Reader) Read(p []byte) (n int, err error) {
+	if lr == nil {
+		return 0, errors.New("nil reader")
```

Comment:
Please define exportable error variable `ErrNilReader` like this:
```
    ErrNilReader = errors.New("nil reader")
```

and use it for all functions that returns an error